src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -65,18 +56,25 @@ public static synchronized String updateSourceCredentials(BuildConfig buildConfi
     return id;
   }
 
+  // getCurrentToken returns the ServiceAccount token currently selected by the user. A return value of empty string
+  // implies no token is configured.
   public static String getCurrentToken() {
+    String credentialsId = GlobalPluginConfiguration.get().getCredentialsId();
+    if(credentialsId == "") {
+      return "";
+    }
```

Comment:
have you done any testing to confirm what value you get back when there *is* a credential configured?